### ISSUE TITLE
Add new validator API for voluntary exit

### ISSUE THREAD

**Issue by jimmygchen** [sigp/lighthouse#4117]
## Description

Add a new validator API to publish voluntary exit. This will make voluntary exit via the [Siren UI](https://github.com/sigp/siren) much easier for the users.

NOTE: this will only work for users that uses the Lighthouse validator client.

### Security

This endpoint is potentially dangerous and would need some added security and barrier to initiate the exit. This will mainly be triggered from the Siren UI.

Potentially, a combination of:
- the existing VC auth scheme
- requiring a flag on the VC to enable it (`--allow-voluntary-exit`)
- takes a JSON body rather than just a POST `/validator/voluntary_exits/<pubkey>`

### Proposed API

| Property | Specification |
| ---------- | ------------- |
| Path | `/lighthouse/validators/voluntary_exits` |
| Method | POST |
| Required Headers | [Authorization](https://lighthouse-book.sigmaprime.io/api-vc-auth-header.html) |
| Typical Responses | 200 | 

#### Example Request Body

```json
{
    "pubkey": "0x93247f2209abcacf57b75a51dafae777f9dd38bc7053d1af526f220a7489a6d3a2753e5f3e8b1cfe39b56f43611df74a"
}
```

#### Error responses

- 401: invalid authorization  key
- 400: invalid public key / request
- 500: internal server error

**Comment by michaelsproul** [comment]
This would be a good candidate for the standard keymanager APIs, seeing as we're going to the trouble of designing and implementing it:

https://github.com/ethereum/keymanager-APIs

**Comment by jimmygchen** [comment]
> This would be a good candidate for the standard keymanager APIs, seeing as we're going to the trouble of designing and implementing it:
> 
> https://github.com/ethereum/keymanager-APIs

Yeah good point, I'll raise an issue in the keymanager API repo. Thanks @michaelsproul

**Comment by jimmygchen** [comment]
Raised issue on the standard key manager APIs repo: https://github.com/ethereum/keymanager-APIs/issues/57

**Comment by jimmygchen** [comment]
Discussed with @AgeManning today and will add the BN submission back to the endpoint. This would be easiest since VC already checks that BN matches the same network. 

I've removed the JSON response, this endpoint would return a status code 200 on success and a few kind of failures mentioned in description above.

I'll be adding a `allow-voluntary-exit` flag (disabled by default) given that this is now a dangerous operation.

**Comment by jimmygchen** [comment]
Implemented in #4119

The endpoint returns a `SignedVoluntaryExit` object, which can be submitted to the [Beacon API](https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Beacon/submitPoolVoluntaryExit).

API spec here: https://ethereum.github.io/keymanager-APIs/?urls.primaryName=dev#/Voluntary%20Exit/signVoluntaryExit
